### PR TITLE
[IMPROVE] ChatBox Text to File Description

### PR DESCRIPTION
**Pull request by eduardofcabrera**
## Proposed changes (including videos or screenshots)

The text content from chatbox goes to the file description when drag and drop a file.

## Steps to test or reproduce

https://user-images.githubusercontent.com/64327259/153050138-73afc945-fdad-46c3-8859-6f559815e199.mp4

### CODE REVIEW

File: app/ui/client/lib/fileUpload.js [hunk 1]
```diff
@@ -118,6 +119,13 @@ export const fileUpload = async (files, input, { rid, tmid }) => {
 		tmid = replies[0]._id;
 	}
 
+	const key = ['messagebox', rid, tmid].filter(Boolean).join('_');
+	let messageBoxText = Meteor._localStorage.getItem(key);
+
+	if (!messageBoxText) {
+		messageBoxText = '';
+	}
```
Comment (ostjen) [inline app/ui/client/lib/fileUpload.js:127]: you can use the double pipe operator here
`let messageBoxText = Meteor._localStorage.getItem(key) || '';`

**Review by ostjen — CHANGES_REQUESTED** [review]

**Comment by ostjen** [comment]
looks promising, just need to keep the text when the upload action is cancelled

File: app/ui/client/views/app/room.js [hunk 2]
```diff
@@ -633,6 +633,9 @@ Meteor.startup(() => {
 		this.tabBar = this.data.tabBar;
 
 		this.onFile = (filesToUpload) => {
+			const { input } = chatMessages[RoomManager.openedRoom];
+			input.value = null;
+			$(input).change().trigger('input');
```
[the remark this answers is not in the document]
Comment (eduardofcabrera) [reply app/ui/client/views/app/room.js:638]: To erease the content from the chat box and reload de icons to send a file or send an audio.

File: app/ui/client/lib/fileUpload.js (hunk 1 above)
Comment (eduardofcabrera) [reply app/ui/client/lib/fileUpload.js:127]: Ok, I'll do this

**Review by ostjen — APPROVED** [review]
tested here and LGTM, but I'll ask someone from the frontend team to review 😃

File: client/views/room/modals/FileUploadModal/FileUploadModal.tsx [hunk 3]
```diff
@@ -11,12 +11,20 @@ type FileUploadModalProps = {
 	onSubmit: (name: string, description?: string) => void;
 	file: File;
 	fileName: string;
+	fileDescription: string;
```
Comment (dougfabris) [inline client/views/room/modals/FileUploadModal/FileUploadModal.tsx:14]:
```suggestion
	fileDescription?: string;
```

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: app/ui/client/lib/fileUpload.js [hunk 4]
```diff
@@ -140,6 +146,12 @@ export const fileUpload = async (files, input, { rid, tmid }) => {
 						msg: msg || undefined,
 						file,
 					});
+					const localStorageKey = ['messagebox', rid, tmid].filter(Boolean).join('_');
+					const chatMessageKey = [rid, tmid].filter(Boolean).join('-');
+					const { input } = chatMessages[chatMessageKey];
+					input.value = null;
+					$(input).change().trigger('input');
```
Comment (dougfabris) [inline app/ui/client/lib/fileUpload.js:153]:
```suggestion
					$(input).trigger('input');
```
Deprecated

**Review by dougfabris — CHANGES_REQUESTED** [review]

**Review by dougfabris — APPROVED** [review]

**Pull request merged**